app/models/ability.rb

```diff
@@ -73,6 +86,21 @@ def initialize(user)
       can? :download, f
     end
 
+    can :read, StacksMediaStream do |f|
+      user_in_location, _rule = f.location_rights(user.location)
+      f.restricted_by_location? && user_in_location
+    end
```

Comment:
On closer inspection I believe I was miss-reading this.  I see that this is adding the read/stream exceptions from the download case.